### ISSUE TITLE
findAndModify in mongodb 2.6.10 vs 3.0.3

### ISSUE THREAD

**Issue by CTAPbIuMABP** [Automattic/mongoose#3052]
npm versions
mongodb 2.0.33
mongoose 4.0.4

js code has different behavior across different mongodb versions

```js
var myModel = mongoose.model("model", new Schema({
	title: String,
	single: {
		type: Schema.Types.ObjectId,
		ref: "anotherModel"
	},
	array: [{
		type: Schema.Types.ObjectId,
		ref: "anotherModel"
	}]
}, {versionKey: false}));

var ObjectId = mongoose.Types.ObjectId();

myModel.create({
	title: "title",
	single: ObjectId,
	array: [ObjectId]
}, function (model) {
	// model has single and array set to ObjectId in both mongodbs
	myModel.findByIdAndUpdate(model._id, {title: "new title"}, function(model){
		// in mongodb 3.0.3 
		// model has no single and empty array field
		// in mongodb 2.6.10
		// model has both fields
		myModel.findById(model._id, function(model){
			// model has both fields in both mongodbs
		});
	});
});
```

**Comment by CTAPbIuMABP** [comment]
i have tried lates gh version and it looks fixed however i see no related commits

**Comment by vkarpov15** [comment]
Weird. Glad it seems fixed with the upcoming 4.0.5 release, but I'll double check later in case there's still an issue.

**Comment by vkarpov15** [comment]
Hmm so given the below script:

```javascript
var mongoose = require('mongoose');
var Schema = mongoose.Schema;
mongoose.set('debug', true);
mongoose.connect('mongodb://localhost:27017/gh2860');

var myModel = mongoose.model("model", new Schema({
    title: String,
    single: {
        type: Schema.Types.ObjectId,
        ref: "anotherModel"
    },
    array: [{
        type: Schema.Types.ObjectId,
        ref: "anotherModel"
    }]
}, {versionKey: false}));

var ObjectId = mongoose.Types.ObjectId();

myModel.create({
    title: "title",
    single: ObjectId,
    array: [ObjectId]
}, function (error, model) {
    // model has single and array set to ObjectId in both mongodbs
    myModel.findByIdAndUpdate(model._id, {title: "new title"}, function(error, model){
        console.log(JSON.stringify(model, null, '  '));
        // in mongodb 3.0.3 
        // model has no single and empty array field
        // in mongodb 2.6.10
        // model has both fields
        myModel.findById(model._id, function(model){
            // model has both fields in both mongodbs
        });
    });
});
```

I get the same output in both 2.6.10 and 3.0.3 on mongoose 4.0.4 and 4.0.6. Either way, this works for you in 4.0.5, so I'm gonna close. Re-open if this is still an issue.
